### ISSUE TITLE
namedtuple: Expected string literal for tuple entry name

### ISSUE THREAD

**Issue by laipz8200** [microsoft/pylance-release#1832]
<h3> Environment data </h3>

-   Language Server version: 2021.9.2
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.7.3 64-bit
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: off

<h3> Expected behaviour </h3>

namedtuple should receive any Iterable[str] type data, but it prompts a problem.

The code runs correctly. In fact, it comes from [this website](https://docs.djangoproject.com/en/3.2/topics/db/sql/#executing-custom-sql-directly). Is this a recommended practice?

<h3> Actual behaviour </h3>

`Expected string literal for tuple entry name`

![20210918111446](https://user-images.githubusercontent.com/16485841/133870732-403cc110-8cd8-40b6-9849-c287ba0d842c.png)

<h3> Logs </h3>

Python Language Server Log
```
[Info  - 10:11:47 AM] Pylance language server 2021.9.2 (pyright eca6f1a5) starting
[Info  - 10:11:47 AM] Server root directory: /home/xxxxxx/.vscode/extensions/ms-python.vscode-pylance-2021.9.2/dist
```

<h3> Code Snippet / Additional information </h3>

```python
def namedtuplefetchall(cursor):
    "Return all rows from a cursor as a namedtuple"
    desc = cursor.description
    nt_result = namedtuple('Result', [col[0] for col in desc])
    nt_result = namedtuple('Result', (col[0] for col in desc))
    return [nt_result(*row) for row in cursor.fetchall()]
```

**Comment by erictraut** [comment]
You're right, this should not generate an error. If you dynamically specify the entries of a namedtuple, the static type checker won't be able to reason about the names and types of the entries, but it's a legitimate use of namedtuple.

I've updated the code to eliminate the false positive error. It will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202193-22-september-2021
